pkg/parser/ast/ddl.go

```diff
@@ -828,17 +828,20 @@ func (n *IndexOption) Restore(ctx *format.RestoreCtx) error {
 		if hasPrevOption {
 			ctx.WritePlain(" ")
 		}
-		ctx.WriteKeyWord("PRE_SPLIT_REGIONS")
-		ctx.WritePlain(" = ")
-		if n.SplitOpt.Num != 0 && len(n.SplitOpt.Lower) == 0 {
-			ctx.WritePlainf("%d", n.SplitOpt.Num)
-		} else {
-			ctx.WritePlain("(")
-			if err := n.SplitOpt.Restore(ctx); err != nil {
-				return errors.Annotate(err, "An error occurred while splicing IndexOption SplitOpt")
+		_ = ctx.WriteWithSpecialComments(tidb.FeatureIDPresplit, func() error {
```

Comment:
```suggestion
		return ctx.WriteWithSpecialComments(tidb.FeatureIDPresplit, func() error {
```
and remove the trailing `return nil`